src/Microsoft.Health.SqlServer/Features/Health/SqlServerHealthCheck.cs

```diff
@@ -62,17 +49,25 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false);
 
             _logger.LogInformation("Successfully connected to SQL database.");
+
             return HealthCheckResult.Healthy("Successfully connected.");
         }
-        // Error: Can not connect to the database in its current state. This error can be for various DB states (recovering, inacessible) but we assume that our DB will only hit this for Inaccessible state
-        catch (SqlException ex) when (ex.ErrorCode == 40925)
+        // Filter on error codes for azure key vault https://learn.microsoft.com/en-us/sql/relational-databases/errors-events/database-engine-events-and-errors-31000-to-41399?view=sql-server-ver16
+        catch (Exception ex) when (ex is SqlException sqlException && (sqlException.ErrorCode is 40981 or 33183 or 33184 or 40925))
```

Comment:
nit: just do something like:
```suggestion
        catch (SqlException e) when (e.ErrorCode is 40981 or 33183 or 33184 or 40925)
```